test/functional/page_objects/dashboard_page.js

```diff
@@ -176,9 +176,16 @@ export function DashboardPageProvider({ getService, getPageObjects }) {
     async switchToEditMode() {
       log.debug('Switching to edit mode');
       await testSubjects.click('dashboardEditMode');
-      await retry.waitFor('not in view mode', async () => (
-        !await this.getIsInViewMode()
-      ));
+      await retry.waitFor('in edit mode', () => this.getIsInEditMode());
+    }
+
+    async getIsInEditMode() {
+      log.debug('getIsInEditMode');
```

Comment:
We might want to add a little comment / note about why we have this and why it isn't just a not of `getIsInViewMode`...

Reply:
I'm wondering if I should change it like this so we wouldn't need any comments?
```
switchToEditMod()
  click('dashboardEditMode')
  waitUntilInEditMode

waitUntilInEditMode()
  waitFor (
     const [panels, menuIcons] = await Promise.all([
        testSubjects.findAll('dashboardPanel'),
        testSubjects.findAll('dashboardPanelToggleMenuIcon'),
      ]);

      return panels.length === menuIcons.length;
```

Reply:
Or actually, the more I think about it, why shouldn't that all just be one method?  When I `switchToEditMode` it should just do it and return when it's done.  Splitting it out only makes sense if we think we're going to use `getIsInEditMode` or `waitUntilInEditMode` by itself for some reason.

Reply:
Also makes me wonder if `clickCancelOutOfEditMode` needs some similar waitFor check...

Reply:
```
    async switchToEditMode() {
      log.debug('Switching to edit mode');
      await testSubjects.click('dashboardEditMode');
      // wait until the count of dashboard panels equals the count of toggle menu icons
      await retry.waitFor('in edit mode', async () => {
        const [panels, menuIcons] = await Promise.all([
          testSubjects.findAll('dashboardPanel'),
          testSubjects.findAll('dashboardPanelToggleMenuIcon'),
        ]);
        return panels.length === menuIcons.length;
      });
    }
```